isbnbot/normalize_isbns.py

```diff
@@ -128,66 +76,66 @@ def run(self, dump_filepath: str) -> None:
                         )
                         self.save(lambda: edition.save(comment=comment))
 
-    def save(self, save_fn):
-        """Modify default save behavior based on 'limit' and 'dry_run' parameters"""
-        if not self.dry_run:
-            save_fn()
-        else:
-            self.logger.info("Modification not made because dry_run is True")
-        self.changed += 1
-        if self.limit and self.changed >= self.limit:
-            self.logger.info("Modification limit reached. Exiting script.")
-            sys.exit()
-
-
-def str2bool(value):
-    if isinstance(value, bool):
-        return value
-    if value.lower() in ("yes", "true", "t", "y", "1"):
-        return True
-    elif value.lower() in ("no", "false", "f", "n", "0"):
-        return False
-    else:
-        raise argparse.ArgumentTypeError("Boolean value expected.")
-
 
-def dedupe(input: list) -> list:
+def dedupe(input_list: list) -> list:
     """Remove duplicate elements in a list and return the new list"""
     output = list()
-    for i in input:
+    for i in input_list:
         if i not in output:
             output.append(i)
     return output
 
 
+def parse_isbns(string: str) -> list:
+    """Find isbns in a string on the assumption that if you strip all non-isbn
+    characters and all that is left is valid isbns then it's unlikely to be random chance
+    if that doesn't work it will use a reasonably strict regex to look for an isbn13
+    formatted string within the full text, which should help when other numbers are present"""
+
+    # pass one: strip all non isbn characters, and see if what remains looks like an ISBN
+    isbnchars = [c for c in string if c in "0123456789Xx"]
+    if len(isbnchars) < 10:
+        return []
+
+    isbnchars = "".join(isbnchars).upper()
+    # X is tricky, but can only appear at the end of an isbn10 so remove if not where expected
+    x_idx = isbnchars.find("X")
+    if (x_idx + 1) % 10 != 0 and x_idx + 1 != 23:
```

Comment:
Why 23?

Reply:
If `string` contains an ISBN10 and an ISBN13 it will be 23 characters long. So the 23rd character could be `X`.

Now that I'm thinking, a more robust check is
```suggestion
    if (x_idx + 1) % 10 != 0 and (x_idx + 1) % 23 != 0:
```